src/controllers/facilitiesController.js

```diff
@@ -82,5 +83,96 @@ class FacilitiesController {
+    try {
+      const { user } = req;
+      const { id } = req.query;
+
+      const likedFacility = await db.Facilities.findOne({ where: { id } });
+      const { likesId, unlikesId } = likedFacility;
+      const alreadyLiked = likesId.find((x) => x === user.id);
+      const unliked = unlikesId.find((d) => d === user.id);
+
+      const unlikedUser = unlikesId.indexOf(unliked);
+      let updateIds;
+
+      if (unliked === user.id && unlikesId.length <= 1) {
+        updateIds = unlikesId.splice(unlikedUser, 0);
+        await db.Facilities.update({ unlikesId: updateIds }, { where: { id } });
+      } else {
+        updateIds = unlikesId.splice(unlikedUser, 1);
+        await db.Facilities.update({ unlikesId: updateIds }, { where: { id } });
+      }
+
+      if (user.id === unliked) {
+        await db.Facilities.decrement('unlikes', { where: { id } });
+      }
+
+      if (alreadyLiked === undefined) {
+        await db.Facilities.increment('likes', { where: { id } });
+        await db.Facilities.update({
+          likesId: Sequelize.fn('array_append', Sequelize.col('likesId'), user.id)
+        }, { where: { id } });
+        return Response.success(res, 200, res.__('user has liked facility'));
+      }
+      return Response.errorResponse(res, 403, res.__('user has already liked facility'));
+    } catch (error) {
+      return Response.errorResponse(res, 500, res.__('server error'));
+    }
+  }
+
+  /**
+    * @description unlike a facility method
+    * @static
+    * @param {Object} req
+    * @param {Object} res
+    * @returns {Object} Room
+    * @memberof FacilitiesController
+  */
+  static async unlikeFacility(req, res) {
+    try {
+      const { user } = req;
+      const { id } = req.query;
+      const unlikedFacility = await db.Facilities.findOne({ where: { id } });
+      const { unlikesId, likesId } = unlikedFacility;
+      const alreadyUnliked = unlikesId.find((u) => u === user.id);
+      const liked = likesId.find((d) => d === user.id);
+
+      const likedUser = likesId.indexOf(liked);
+      let updateIds;
+
+      if (liked === user.id && likesId.length <= 1) {
+        updateIds = likesId.splice(likedUser, 0);
+        await db.Facilities.update({ likesId: updateIds }, { where: { id } });
+      } else {
+        updateIds = likesId.splice(likedUser, 1);
+        await db.Facilities.update({ likesId: updateIds }, { where: { id } });
+      }
+
+      if (user.id === liked) {
+        await db.Facilities.decrement('likes', { where: { id } });
+      }
+
+      if (alreadyUnliked === undefined) {
+        await db.Facilities.increment('unlikes', { where: { id } });
+        await db.Facilities.update({
+          unlikesId: Sequelize.fn('array_append', Sequelize.col('unlikesId'), user.id)
+        }, { where: { id } });
+        return Response.success(res, 200, res.__('user has unliked facility'));
+      }
+      return Response.errorResponse(res, 403, res.__('user has already unliked facility'));
```

Comment:
hey @Ntare22, shouldn't the unlike be removed when the user tries to unlike it two times